node/node.go

```diff
@@ -90,3 +91,28 @@ func (s *Service) Delete(ctx context.Context, name string) error {
 	}
 	return nil
 }
+
+// DeleteCollection deletes all nodes.
+func (s *Service) DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error {
+	ns, err := s.client.CoreV1().Nodes().List(ctx, lopts)
+	if err != nil {
+		return xerrors.Errorf("list nodes: %w", err)
+	}
+
+	for _, n := range ns.Items {
+		// delete pods on specific node
+		lopts := metav1.ListOptions{
+			FieldSelector: "spec.nodeName=" + n.Name,
+		}
+		if err := s.podService.DeleteCollection(ctx, lopts); err != nil {
+			return xerrors.Errorf("failed to delete pods on node %s: %w", n.Name, err)
+		}
+
+		// delete specific node
+		if err := s.client.CoreV1().Nodes().Delete(ctx, n.Name, metav1.DeleteOptions{}); err != nil {
+			return xerrors.Errorf("delete node: %w", err)
+		}
```

Comment:
process these part in parallel with errgroup or waitgroup

Reply:
Thanks ! I agree with you.